osbs/api.py

```diff
@@ -726,28 +734,58 @@ def import_image(self, name):
 
     @osbsapi
     def get_token(self):
-        return self.os.get_oauth_token()
+        if self.os.use_kerberos:
+            return self.os.get_oauth_token()
+        else:
+            if self.os.token:
+                return self.os.token
+
+            raise OsbsValidationException("no token stored")
 
     @osbsapi
-    def login(self, token):
-        token_file = utils.get_instance_token_file_name(self.os_conf.conf_section)
-        token_file_dir = os.path.dirname(token_file)
+    def login(self, token, username, password):
+        if self.os.use_kerberos:
+             raise OsbsValidationException("can't use login when using kerberos")
+
+        if not token:
+            if username:
+                self.os.username = username
+            else:
+                try:
+                    self.os.username = raw_input("Username: ")
+                except NameError:
+                    self.os.username = input("Username: ")
+
+            if password:
+                self.os.password = password
+            else:
+                self.os.password = getpass.getpass()
+            self.os.use_auth = True
+            token = self.os.get_oauth_token()
+
+        if token:
+            self.os.token = token
+            if self.os.get_user_login_test().status_code == httplib.UNAUTHORIZED:
+                raise OsbsValidationException("token is not valid")
```

Comment:
Is it important for this to raise OsbsValidationException instead of OsbsResponseException for some reason?

If so, why not do this?:
```
try:
    self.os.get_user()
except OsbsResponseException as ex:
    if ex.status_code == httplib.UNAUTHORIZED:
        raise OsbsValidationException(...)
    raise
```

It doesn't need a new core method.